src/pyrokinetics/gk_code/gene.py

```diff
@@ -416,62 +411,63 @@ def get_local_geometry(self) -> LocalGeometry:
         if geometry_type in ["tracer_efit", "gene"]:
             lref = local_geometry.Rmaj.units
             bref = local_geometry.B0.units
-
-            local_geometry.R_eq = geometry_dict["R_eq"] * lref
-            local_geometry.Z_eq = geometry_dict["Z_eq"] * lref
-            local_geometry.Rmaj = geometry_dict["Rmaj"] * lref
-            local_geometry.b_poloidal_eq = geometry_dict["b_poloidal_eq"] * bref
-            local_geometry.Z0 = geometry_dict["Z0"] * lref
-            local_geometry.rho = geometry_dict["rho"] * lref
-            local_geometry.dpsidr = geometry_dict["dpsidr"] * bref * lref
-
-            local_geometry.beta_prime = geometry_dict["beta_prime"] * bref**2 / lref
-
             self._drhotor_dr = geometry_dict["drhotor_dr"]
 
-            local_geometry._set_shape_coefficients(
-                local_geometry.R_eq,
-                local_geometry.Z_eq,
-                local_geometry.b_poloidal_eq,
+            # Create a mock LocalGeometry using geometry dict and fit to it.
+            # This will get a new set of shaping parameters, which are then modified.
+
+            mock = LocalGeometryFourierGENE.from_local_geometry(
+                LocalGeometry(
+                    psi_n=local_geometry.psi_n,
+                    rho=geometry_dict["rho"] * lref,
+                    a_minor=local_geometry.a_minor,
+                    Rmaj=geometry_dict["Rmaj"] * lref,
+                    Z0=geometry_dict["Z0"] * lref,
+                    Fpsi=local_geometry.Fpsi,
+                    FF_prime=local_geometry.FF_prime,
+                    B0=local_geometry.B0,
+                    bunit_over_b0=local_geometry.bunit_over_b0,
+                    dpsidr=geometry_dict["dpsidr"] * bref * lref,
+                    q=local_geometry.q,
+                    shat=geometry_dict["shat"],
+                    beta_prime=geometry_dict["beta_prime"] * bref**2 / lref,
+                    R=geometry_dict["R"] * lref,
+                    Z=geometry_dict["Z"] * lref,
+                    b_poloidal=geometry_dict["b_poloidal"] * bref,
+                    theta=geometry_dict["theta"],
+                    bt_ccw=local_geometry.bt_ccw,
+                    ip_ccw=local_geometry.ip_ccw,
+                    dRdtheta=local_geometry.dRdtheta,
+                    dRdr=local_geometry.dRdr,
+                    dZdtheta=local_geometry.dZdtheta,
+                    dZdr=local_geometry.dZdr,
+                )
```

Comment:
I've finished merging and attempted to recreate the new GENE features using my new methods, but I can't tell if I've done so correctly or not. This section of the code looks like it's untested, and I don't really know what it's doing.

Reply:
Essentially the full data needed to populate `LocalGeometry` is contained inside a different file which requires a bit of post-processing to get the data in a form that can actually be read by `LocalGeometry`. What you do here looks to be correct. This is tested here https://github.com/pyro-kinetics/pyrokinetics/blob/765d0f1032470937926431847a9f87efefcdcd41/tests/local_geometry/test_fourier.py#L383

Reply:
That test doesn't seem to touch the new code in `gk_code/gene.py`, so I'm not fully confident that my changes haven't broken something. I can leave it alone for now since it's slightly outside the scope of the PR, but it would be good to get a test in that uses one of those new geometry files alongside a GENE input file directly.

Reply:
Right I see what you mean! I just fit a `FourierGENE` to the GEQDSK and compare to a new geometry file I made independently. I can add that in but yes its outside the scope of this PR